fortitude/src/check.rs

```diff
@@ -323,17 +328,21 @@ fn get_files<P: AsRef<Path>, S: AsRef<str>>(
     extensions: &[S],
     excludes: &FilePatternSet,
     exclude_mode: ExcludeMode,
+    gitignore_mode: GitignoreMode,
 ) -> Vec<PathBuf> {
     paths
```

Comment:
Is it not possible to build the walker at this level? Then we can avoid building a new walker for each individual directory. It might also make it easier to switch to the parallel walker in the future

Reply:
I tried to do it this way initially but ran into a bunch of problems getting it to work properly. I can give it another go.

Reply:
https://github.com/PlasmaFAIR/fortitude/pull/285/commits/7869a739436198483392a181ff349aa812cc3884 implements this. It makes the function a fair bit messier, as invalid paths need to be handled separately. `WalkBuilder` will raise an error if it's given a path that doesn't exist, but we need them to persist (either that or we rewrite the whole way we're handling missing files, but as there are already outstanding bugs related to the reporting of missing files I'd rather keep things the way they are in this PR if we can).